### PR TITLE
hotfix: only show vfolder when mount in session perm is allowed

### PR DESCRIPTION
**Pull request by lizable**
> NOTE: need to test in Backend.AI Core which adopted this PR changes
- Related: [backend.ai#2056](https://github.com/lablup/backend.ai/pull/2056)

This PR resolves the issue in Neo session launcher that shows vfolder list regardless of "`mount-in-session`" permission. For better user experience, we need to show vfolder which has proper permission unless It will throw error like this:

https://github.com/lablup/backend.ai-webui/assets/46954439/cfa97c2a-3253-4368-8ca2-a745063de041

**Checklist:** (if applicable)

- [ ] Mention to the original issue
- [ ] Documentation
- [ ] Minimum required manager version
- [ ] Specific setting for review (eg., KB link, endpoint or how to setup)
- [ ] Minimum requirements to check during review
- [ ] Test case(s) to demonstrate the difference of before/after

### CODE REVIEW

File: react/src/components/VFolderTable.tsx [hunk 1]
```diff
@@ -132,8 +143,61 @@ const VFolderTable: React.FC<VFolderTableProps> = ({
...
+    allowedVFolderHostsByDomain,
+    allowedVFolderHostsByGroup,
+    allowedVFolderHostsByKeypairResourcePolicy,
+  );
+  // only allow mount if volume permission has 'mount-in-session'
+  const mountAllowedVolumes = Object.keys(mergedVFolderPermissions)
+    .filter((volume) =>
+      mergedVFolderPermissions[volume].includes('mount-in-session'),
+    )
+    .map((volume) => volume);
```
Comment (yomybaby) [inline react/src/components/VFolderTable.tsx]: We can remove this. ` .map((volume) => volume);`

File: react/src/components/VFolderTable.tsx [hunk 2]
```diff
@@ -132,8 +143,61 @@ const VFolderTable: React.FC<VFolderTableProps> = ({
...
+      {
+        fetchPolicy: 'store-and-network',
+      },
+    );
+
+  const allowedVFolderHostsByDomain = JSON.parse(domain?.allowed_vfolder_hosts);
+  const allowedVFolderHostsByGroup = JSON.parse(group?.allowed_vfolder_hosts);
+  const allowedVFolderHostsByKeypairResourcePolicy = JSON.parse(
+    keypair_resource_policy?.allowed_vfolder_hosts,
+  );
```
Comment (agatha197) [inline react/src/components/VFolderTable.tsx]:
```suggestion
  const allowedVFolderHostsByDomain = JSON.parse(domain?.allowed_vfolder_hosts || '{}');
  const allowedVFolderHostsByGroup = JSON.parse(group?.allowed_vfolder_hosts || '{}');
  const allowedVFolderHostsByKeypairResourcePolicy = JSON.parse(
    keypair_resource_policy?.allowed_vfolder_hosts || '{}',
  );
```
If `allowed_vfolder_hosts` is undefined, you can see this error.
image

**Review by agatha197 — CHANGES_REQUESTED** [review]

**Review by yomybaby — CHANGES_REQUESTED** [review]

**Review by agatha197 — APPROVED** [review]
LGTM!

File: react/src/components/VFolderTable.tsx [hunk 3]
```diff
@@ -169,12 +169,18 @@ const VFolderTable: React.FC<VFolderTableProps> = ({
...
       },
       {
-        fetchPolicy: 'store-and-network',
+        fetchPolicy: 'network-only',
       },
     );
 
-  const allowedVFolderHostsByDomain = JSON.parse(domain?.allowed_vfolder_hosts || '{}');
-  const allowedVFolderHostsByGroup = JSON.parse(group?.allowed_vfolder_hosts || '{}');
+  console.log(domain);
```
Comment (yomybaby) [inline react/src/components/VFolderTable.tsx]: Please remove this

Comment (yomybaby) [reply react/src/components/VFolderTable.tsx]:
```suggestion
```

**Review by yomybaby — CHANGES_REQUESTED** [review]

**Comment by yomybaby** [comment]
@lizable Please check a review and resolve conflicts. 😉

**Review by yomybaby — APPROVED** [review]
LGTM

**Pull request merged**